src/chttpd/src/chttpd_db.erl

```diff
@@ -1242,6 +1242,14 @@ bulk_get_multipart_boundary() ->
 receive_request_data(Req) ->
     receive_request_data(Req, chttpd:body_length(Req)).
 
+receive_request_data(Req, Len) when Len == chunked ->
+    ChunkFun = fun({_Length, Binary}, _State) ->
+        self() ! {chunk, Req, Binary}
```

Comment:
Done in 9bd0e936e .

> Was it done to match and assert later one that we only receive the message from the same request? 

Yes. Adding `Req` (or a ref) should not be necessary, it's just more of a defensive programming precaution thing.

> I wonder if we need to pass the `Req` object around here for every chunk read as it's quite large.

I'm no erlang expert, but I thought since we're in the same process, this would be a reference passed anyway (and not a value / copy) so the size of `Req` would not matter.

Reply:
Good question. It is sent in a message, most of those terms (except large binaries and literals) are copied. But maybe there is an optimization I hadn't heard of to avoid a copy if the message is sent to the same process...